src/promptflow-devkit/promptflow/_sdk/_service/apis/collector.py

```diff
@@ -134,18 +135,33 @@ def _try_write_trace_to_cosmosdb(
         )
         collection_client_thread.start()
 
+        line_summary_client_thread = ThreadWithContextVars(
+            target=get_client,
+            args=(CosmosDBContainerName.LINE_SUMMARY, subscription_id, resource_group_name, workspace_name, credential),
+        )
+        line_summary_client_thread.start()
+
         # Load created_by info first time may slow. So, we load it in parallel for warm up.
         created_by_thread = ThreadWithContextVars(target=get_created_by_info_with_cache)
         created_by_thread.start()
 
-        get_client(CosmosDBContainerName.LINE_SUMMARY, subscription_id, resource_group_name, workspace_name, credential)
+        # Get default blob may be slow. So, we have a cache for default datastore.
+        from promptflow.azure._storage.blob.client import get_client as get_blob_client
+
+        blob_container_client, blob_base_uri = get_blob_client(
```

Comment:
I know there is cache for container client. So we do not use thread here? What's the purpose of using thread for cosmos cilent but not using thread for container client?

Reply:
The logic of get blob client run at main thread, not need to start a new thread.